tests/test_single_topology_v3.py

```diff
@@ -383,3 +394,131 @@ def test_no_chiral_bond_restraints():
     assert len(state.chiral_bond.get_idxs()) == 0
     U = state.get_U_fn()
     _ = U(init_conf)
+
+
+def assert_interpolation_valid(f, a, b):
+    np.testing.assert_allclose(f(a, b, 0.0), a)
+    np.testing.assert_allclose(f(a, b, 1.0), b)
+
+    rng = np.random.default_rng(2022)
+    lambdas = rng.uniform(0, 1, (30,))
+    ys = vmap(functools.partial(f, a, b))(lambdas)
+    assert np.all(np.minimum(a, b) <= ys)
+    assert np.all(ys <= np.maximum(a, b))
+
+
+def test_handle_ring_opening_closing():
```

Comment:
Sorry if I missed it, but also add test asserting vacuum stability on hif2a dataset examples of ring opening/closing?

Reply:
Good call, I had not done this yet but will add.

Reply:
can we give you a couple of more test-cases of the form:

mol_a_sdf, mol_b_sdf, core_ab?

Reply:
oops, apparently I'd accidentally marked this as resolved. Still in progress..

Reply:
> Sorry if I missed it, but also add test asserting vacuum stability on hif2a dataset examples of ring opening/closing?

This is so far seeming a bit tricky -- the cases I've uncovered so far seem to have pretty low probability / step of blowing up (i.e. require ~100k steps to blow up), so it seems like we'd need a lot of samples to be reasonably sure that we've eliminated the instabilities. Thinking of a couple options to test this:

1. Start from a seed and initial conformation that's known to blow up in a small number of steps; assert that it does blow up when we interpolate bonds/angles/torsions simultaneously, but is safe with sequencing. Downside: if we rely on a seed, this wouldn't be robust to minor code changes.

2. Test that assertions of the form `k(i, j, k, l) > 0 => k(i, j, k) > k_min and k(j, k, l) > k_min` hold (where `k_min` is the minimum safe angle force constant for enabling torsions)

Reply:
Okay - let's set up the examples in a separate PR then. Don't want to block the PR.

Reply:
(Ideally I still want to do something that involves sampling - but this may be a bit tricky as you've noted)

Reply:
Created #850 to track